docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,264 @@
+
+## 3. Proposed Commands
+**Note**: In the context of version 1.4, currently under development, no changes to the command-line interface should be
+allowed. Therefore, we also propose alternative approaches to run the same RAG pipelines using existing `ilab` commands or
+other provided tools.
+
+### 3.1 RAG Ingestion Pipeline Command
+The proposal is to add a `rag` subgroup under the `data` group, with an `ingest` command, like:
+```
+ilab data rag ingest /path/to/docs/folder
+```
+
+#### Command Purpose
+This command processes embeddings generated from documents located in the */path/to/docs/folder* folder and stores them in a vector database. These embeddings are intended to be used as augmented context in a Retrieval-Augmented Generation (RAG) chat pipeline.
+
+#### Assumptions
+The documents must be in JSON format and pre-processed using the `docling` tool.
+
+**Note**: The expected JSON schema is the same that we have when we generate knowledge data with `ilab data generate` and we
+add a reference document to the `qna.yaml` document(s).
+
+#### Supported Databases
+The command supports multiple vector database types. By default, it uses a local `MilvusLite` instance stored at `./rag-output.db`.
+
+#### Usage
+The generated embeddings can later be retrieved to enrich the context for RAG-based chat pipelines.
+
+#### Running the pipeline with v1.4
+Currently, there is no `ilab data rag` command available to execute the RAG ingestion pipeline. As a result, it is not possible
+to address the limitations of version 1.4 using the configuration options detailed below. 
+Alternatively, we propose providing a Jupyter notebook or a standalone script to run the pipeline with default settings.
+
+The script should guide users in overriding default options and generating the RAG artifacts in the configured vector database instance. By default, support for MilvusLite will be included.
+
+> **ℹ️ CLI-based alternative:** This is an alternative to run both the document transformation and the document ingestion in 
+> could be to run the document ingestion pipeline as part of the existing
+> `ilab data generate` command, with additional options to:
+> * overcome the limitation of starting from `qna.yaml` documents
+> * define an input and output locations
+> Example:
+> ```yaml
+> generate:
+>   rag: 
+>     enabled: true
+>     input: /path/to/source-docs/folder
+>     output: /path/to/processed-docs/folder
+> rag:
+>   ...
+> ```
+> When `generate.rag.enabled` is set to `true`, the `ilab data generate command` executes the document ingestion pipeline, transforming 
+> the source documents in the `generate.rag.input` folder into the configured vector database:
+> * Step 1: Transform the documents using the existing `SDG` modules (powered by `docling`).
+> * Step 2: Ingest the pre-processed artifacts from `generate.rag.output` into the configured database.
+
+**TODO** Introduce the evaluation framework
```

Comment:
I'd prefer to have a separate dev-doc for evaluation framework because I think it is going to take longer to get that sorted out than some of these more urgent issues.

Reply:
+1

Reply:
Agree, after you latest messages in the Slack discussion channel it's definitely worth parking this point for now.